samples/user-interface/draganddrop/src/main/java/com/example/platform/ui/draganddrop/DragAndDrop.kt

```diff
@@ -0,0 +1,258 @@
+/*
+ * Copyright 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.platform.ui.draganddrop
+
+import android.content.ClipData
+import android.content.ClipDescription.MIMETYPE_TEXT_PLAIN
+import android.content.Intent
+import android.graphics.Bitmap.CompressFormat
+import android.graphics.drawable.BitmapDrawable
+import android.os.Bundle
+import android.os.ParcelFileDescriptor
+import android.util.Log
+import android.util.TypedValue.COMPLEX_UNIT_SP
+import android.view.View
+import android.view.View.DRAG_FLAG_GLOBAL
+import android.view.View.DRAG_FLAG_GLOBAL_URI_READ
+import android.widget.ImageView
+import android.widget.TextView
+import androidx.annotation.RequiresApi
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import androidx.core.content.FileProvider
+import androidx.core.graphics.drawable.toBitmap
+import androidx.core.util.component1
+import androidx.core.util.component2
+import androidx.core.view.DragStartHelper
+import androidx.draganddrop.DropHelper
+import com.example.platform.ui.draganddrop.databinding.DragAndDropActivityBinding
+import com.google.android.catalog.framework.annotations.Sample
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.io.FileOutputStream
+
+@RequiresApi(24)
+@Sample(
+    name = "Drag and Drop",
+    description = "Demonstrates basic Drag and Drop functionality.",
+    documentation = "https://developer.android.com/develop/ui/views/layout/sw-keyboard",
+)
+class DragAndDropActivity : AppCompatActivity() {
+
+    companion object {
+        private const val LOG_TAG = "DragDropSample"
+        private const val MAX_LENGTH = 200
+        fun logE(message: String) {
```

Comment:
Nit: this seems uncommon. Can we remove this

Reply:
Inlined and removed the function.